pkg/controller/rds/dbcluster/setup.go

```diff
@@ -111,34 +109,39 @@ func (e *custom) postObserve(ctx context.Context, cr *svcapitypes.DBCluster, res
 	obs.ConnectionDetails = managed.ConnectionDetails{
 		xpv1.ResourceCredentialsSecretEndpointKey: []byte(aws.StringValue(cr.Status.AtProvider.Endpoint)),
 		xpv1.ResourceCredentialsSecretUserKey:     []byte(aws.StringValue(cr.Spec.ForProvider.MasterUsername)),
-		xpv1.ResourceCredentialsSecretPortKey:     []byte(strconv.FormatInt(aws.Int64Value(cr.Spec.ForProvider.Port), 10)),
 	}
-	pw, _, _ := dbinstance.GetPassword(ctx, e.kube, cr.Spec.ForProvider.MasterUserPasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
-	if pw != "" {
-		obs.ConnectionDetails[xpv1.ResourceCredentialsSecretPasswordKey] = []byte(pw)
+
+	if aws.Int64Value(cr.Spec.ForProvider.Port) > 0 {
+		obs.ConnectionDetails[xpv1.ResourceCredentialsSecretPortKey] = []byte(strconv.FormatInt(aws.Int64Value(cr.Spec.ForProvider.Port), 10))
```

Comment:
Can be simplified by using `Atoi()`

```suggestion
		obs.ConnectionDetails[xpv1.ResourceCredentialsSecretPortKey] = []byte(strconv.Atoi(aws.Int64Value(cr.Spec.ForProvider.Port)))
```